### DISCUSSION TITLE
CAVMN-10, Windows Media Player and 24 bit

### DISCUSSION THREAD

**Discussion by AndrewHinge** [4gra/pclk-mn10#10] [Q&A] [answered]
Hi Graham, I think you are doing some great work here, and you probably more active on learning about M-Crew and the CAVMN-10 than anybody else in 2022.

My question is only distantly related to your coding work here. I am having all kinds of trouble sending 24bit .wav files through the PC Link. I have tried everything, from 24bit sound cards to Windows Media Player (WMP) codec packs.

As I understand it, M-Crew is heavily reliant on WMP to send audio files to the MN10. Therefore, I am on WMP 9 (6.4 is the minimum requirement on the manual) and have added a codec pack that allows playback of 24bit 44.1khz audio files through the computer's inbuilt (bog-standard) sound card. (This is likely emulating 24bit and delivering 16bit).

When I register the 24bit files into M-Crew, there is silence when I try to record, while 16bit files record as normal.

The SPDIF format shouldn't be the limitation here, as it has been proven to handle 24bit stereo. Now that WMP is playing the files, the suspicion now falls to whether M-Crew and the CAVMN-10 are actually able to handle 24bit files (it wouldnt surprise me given the age of the devices).

Am I correct in assuming that the CAV-MN10 is (among other things) an external sound card, and will not benefit from having a PCMCIA 24bit-capable sound card?

Do you think that I should drop my 24bit PC Link dream and stick to 16bit?

**My Specs:**
Sony MXD-D5C (5 x CD, 1 x MD deck), CAVMN-10, Windows XP SP3 32bit, M-Crew 2.01e, 2006 Laptop with AMD chipset. There is an integrated soundcard (dont know the model or specs), I also have a PCMCIA Sound Blaster Audigy 2 Zs Notebook.

By the way, I am happy to test anything you need; while I am not a natural 'coding guy', I would be able to follow instructions well enough. If you want me to try out your code on my 2011 Macbook Pro or the Laptop, I would do my best.

Thanks!

**Comment by 4gra** [comment, marked as answer]
Andrew,

I'm 99% sure the MN10 can't handle anything but 16-bit, 44,100 stereo - even if the chipset can pass 24-bit down the fibre, it's not exposed by the driver so that would need to be reworked.  I also know 24-bit support has been a bit of a contentious area so without wanting to comment on whether it's realistic to do this end-to-end on any given setup... I'll take your word it's possible in theory on your deck.

So, the great news is that you're right - the PCLK/CAV-MN10 *is* effectively a USB (analogue and optical) sound card multiplexed with a serial control bus device (not via a USB hub, but you can sort of think of it that way at a high level). So, you *could* rig a PC -> SPDIF -> MD to send 24-bit audio if you have appropriate output kit, and just plug the 24-bit optical link in place of the MN10's (just make sure you leave the DIN plug in for the rest of the control functionality, and ignore the optical and analogue outputs of the MN10 altogether).

Will that work with M-Crew without modification? Maybe, I've not tested it for such a long time but if you can use the onboard sound (i.e. any alternative sound card), I don't see why not.  But in any event you could "just" record the audio directly without M-Crew - sync recording would just take a few improvements to my toolkit (one day, one day... not possible with what I've got just now but, daydreaming a bit, a VLC plugin using "my" commands would be an example a way to synchronise audio output with MN10 commands.)

**Reply by AndrewHinge** [reply]
Thank you Graham, that's a pretty clear answer, I thank you. I do think I should try the optical output from my 24bit card, I'll let you know the results.
I feel I am almost ready to make my full review video of M-Crew, Ive been sweating over this archaic technology for many months now... hoping for a viable solution for uncompressed audio files straight from artists, have one lossey compression stage (In the machine) and come up with the best possible recording on minidisc, without the pain.
On the topic of recording 20-24bit audio on minidisc, apparently it is possible. I have found an article (Here: http://marlene-d.blogspot.com/2012/06/blast-from-past-minidisc-wide-bit.html) where she uses Sony's 'Wide Bit Stream' technology with Atrac Type-R DSP. She tested the output and sure enough, 20 bit was measured. This is truly tantalising news. I know that many people say there is no point recording in anything higher than 16 bit, however I sure want to try, and see if there is a difference.
By the way, thank you for alerting me to George Styles' M-Crew alternative, Ive found it here on Github and I will test that out too.
All this is culminating in a video Im in the middle of planning that will go in-depth about M-Crew and its limitations. I will be thanking yourself and George Styles for all your hard work on this fun but archaic technology. :)

**Reply by 4gra** [reply]
Sounds like you've been doing some great,  novel work yourself. Thanks for your kind words and for getting in touch, I'll await your video with interest. And, yes, Marlene and George both deserve credit.

**Reply by AndrewHinge** [reply]
> Andrew,
> 
> I'm 99% sure the MN10 can't handle anything but 16-bit, 44,100 stereo - even if the chipset can pass 24-bit down the fibre, it's not exposed by the driver so that would need to be reworked. I also know 24-bit support has been a bit of a contentious area so without wanting to comment on whether it's realistic to do this end-to-end on any given setup... I'll take your word it's possible in theory on your deck.
> 
> So, the great news is that you're right - the PCLK/CAV-MN10 _is_ effectively a USB (analogue and optical) sound card multiplexed with a serial control bus device (not via a USB hub, but you can sort of think of it that way at a high level). So, you _could_ rig a PC -> SPDIF -> MD to send 24-bit audio if you have appropriate output kit, and just plug the 24-bit optical link in place of the MN10's (just make sure you leave the DIN plug in for the rest of the control functionality, and ignore the optical and analogue outputs of the MN10 altogether).
> 
> Will that work with M-Crew without modification? Maybe, I've not tested it for such a long time but if you can use the onboard sound (i.e. any alternative sound card), I don't see why not. But in any event you could "just" record the audio directly without M-Crew - sync recording would just take a few improvements to my toolkit (one day, one day... not possible with what I've got just now but, daydreaming a bit, a VLC plugin using "my" commands would be an example a way to synchronise audio output with MN10 commands.)

Graham, just to report back, last night I tried your solution for bypassing the MN10's optical facilities, in favour of my 24 bit sound card (optical out). The results were interesting.
I still had silence when I tried to play and record from M-Crew. However, when I played the same file in WMP, and set M-Crew's recording source to 'optical', the track recorded no problem. Obviously, without the benefit of synchro recording and track names, this is of limited use to me.
This leads me to believe that, while I'm sure you are correct about the MN10's hardware not being able to handle anything above 16bit, the software seems to not be able to handle 16bit either. Never mind. I read somewhere online that if WMP can play it, M-Crew can too, however this seems not to be the case in this instance. I guess I could test different versions of WMP, however I'm not sure I would have much luck. Right now I feel like I have hit the final wall, and I am happy to settle with 16bit... it's nothing to be sniffed at.
Anyway, this is just FYI. 
Again, if you would like me to test something, let me know.
Thanks again!

**Reply by AndrewHinge** [reply]
Thanks for your help Graham, my video about M-crew is now live! I thank you in the video, you can watch it here > https://youtu.be/Rz2Fps8drdc

**Reply by 4gra** [reply]
Fantastic, and I quite agree with your conclusions! I don't have much time to spend on this project but I am glad there are still a good few people who appreciate what is possible - I still use my PCLK daily with this software as a home automation tool!

**Reply by AndrewHinge** [reply]
> Fantastic, and I quite agree with your conclusions! I don't have much time to spend on this project but I am glad there are still a good few people who appreciate what is possible - I still use my PCLK daily with this software as a home automation tool!

Really? How do you mean? What functionality can it do aside from music?

**Comment by BizzleAnonymous** [comment]
I am a student in IT and its my first year of studies... but guys i need to practice on how to do window installation, but the problem is... i real dont know how to get it for free because im out of budget so any other ideas please share with me... i really like computing so i must or die trying... thanks guys
